HW_1.py

```diff
@@ -0,0 +1,253 @@
+# Task 1
+import datetime
+
+class User:
+    def __init__(self, user):
+        self.user = user
+
+class Message(User):
+    def __init__(self, user, text):
+        self.text = text
+        self.user = user
+        self.datetime = datetime.datetime.now()
+
+    def show(self):
+        #self.datetime = datetime
+        print(f"[{self.datetime}] {self.user}: {self.text}")
+
+    def send(self):
+        return (f"[{self.datetime}] {self.user}: {self.text}")
+
+class Chat(Message):
+    def __init__(self, chat_history):
+        self.chat_history = chat_history
+        self.datetime_history = []
+
+    def recieve(self, message):
+        self.chat_history.insert(0, message)
+
+    def show_last_message(self):
+         print(self.chat_history[0])
+
+# get_history_from_time_period я не реализовала :(
+
+    def show_chat(self):
+        for message in self.chat_history:
+            print(message)
+
+empty_chat_history = []
+chat = Chat(empty_chat_history)
+user1 = User('Scott')
+user2 = User('Ramona')
+
+message1 = Message(user2.user, 'What kind of tea do you want?')
+send_message1 = message1.send()
+chat.recieve(send_message1)
+message1.show()
+
+message2 = Message(user1.user, "There's more than one kind?")
+send_message2 = message2.send()
+chat.recieve(send_message2)
+message2.show()
+
+message3 = Message(user2.user, """We have blueberry, raspberry, ginseng, sleepy time, green tea,
+green tea with lemon, green tea with lemon and honey, liver disaster, ginger with
+honey, ginger without honey, vanilla almond, white truffel, blueberry chamomile, 
+vanilla walnut, constant comment and... earl grey.""")
+send_message3 = message3.send()
+chat.recieve(send_message3)
+message3.show()
+
+message4 = Message(user1.user, "Did you make some of those up?")
+send_message4 = message4.send()
+chat.recieve(send_message4)
+message4.show()
+
+chat.show_chat()
+
+chat.show_last_message()
+
+# Task 2
+class Args:
+    def __init__(self, *args, **kwargs):
+        self.args = args
+        self.kwargs = kwargs
+
+    def __rlshift__(self, f):
+        return f(*self.args, **self.kwargs)
+sum << Args([1, 2])
+(lambda a, b, c: a**2 + b + c) << Args(1, 2, c=50)
+
+# Task 3
+
+class StrangeFloat(float):
+    def __getattr__(self, attr):
+        if "_" in attr:
+            action, number = attr.split("_")
+            if action == "add":
+                return StrangeFloat(self.__add__(float(number)))
```

Comment:
Если есть возможность вместо дандера написать сам оператор, то лучше сделать так. Писать `__add__` может понадобиться разве что в случае `super().__add__`

```suggestion
                return StrangeFloat(self + float(number))
```